repositories/permissions_requests.go

```diff
@@ -0,0 +1,121 @@
+package repositories
+
+import "github.com/topfreegames/Will.IAM/models"
+
+// PermissionsRequests repository
+type PermissionsRequests interface {
+	Clone() PermissionsRequests
+	Create(*models.PermissionRequest) error
+	Deny(string, string) error
+	Get(string) (*models.PermissionRequest, error)
+	Grant(string, string) error
+	ListOpenRequestsVisibleTo(*ListOptions, string) ([]models.PermissionRequest, error)
+	ListOpenRequestsVisibleToCount(string) (int64, error)
+	setStorage(*Storage)
+}
+
+type permissionsRequests struct {
+	*withStorage
+}
+
+func (prs *permissionsRequests) Clone() PermissionsRequests {
+	return NewPermissionsRequests(prs.storage.Clone())
+}
+
+func (prs *permissionsRequests) Create(pr *models.PermissionRequest) error {
+	_, err := prs.storage.PG.DB.Query(
+		pr, `INSERT INTO permissions_requests (service, ownership_level, action, resource_hierarchy,
+    alias, message, state, service_account_id) VALUES (?service, ?ownership_level, ?action,
+    ?resource_hierarchy, ?alias, ?message, ?state, ?service_account_id)
+    ON CONFLICT (service, ownership_level, action, resource_hierarchy, service_account_id)
+    WHERE state = 'open' DO NOTHING RETURNING id`, pr,
+	)
+	return err
+}
+
+func (prs *permissionsRequests) Deny(saID, prID string) error {
+	_, err := prs.storage.PG.DB.Exec(
+		`UPDATE permissions_requests SET state = ?, moderator_service_account_id = ?, updated_at = now()
+    WHERE id = ?`, models.PermissionRequestStates.Denied, saID, prID,
+	)
+	return err
+}
+
+func (prs *permissionsRequests) Get(prID string) (*models.PermissionRequest, error) {
+	var pr models.PermissionRequest
+	if _, err := prs.storage.PG.DB.Query(
+		&pr, ` SELECT * FROM permissions_requests WHERE id = ? `, prID,
+	); err != nil {
+		return nil, err
+	}
+	return &pr, nil
+}
+
+func (prs *permissionsRequests) Grant(saID, prID string) error {
+	_, err := prs.storage.PG.DB.Exec(
+		`UPDATE permissions_requests SET state = ?, moderator_service_account_id = ?, updated_at = now()
+    WHERE id = ?`, models.PermissionRequestStates.Granted, saID, prID,
+	)
+	return err
+}
+
+func (prs *permissionsRequests) ListOpenRequestsVisibleTo(
+	lo *ListOptions, saID string,
+) ([]models.PermissionRequest, error) {
+	var prSl []models.PermissionRequest
+	if _, err := prs.storage.PG.DB.Query(
+		// TODO: this query can probably be optimized if we group saop by service, action, rh and try to
```

Comment:
nit: aim the TODO at an user (the person who has more context about it, not necessarily the person who's fixing it) or at an issue (Github, Jira, what have you). Eg: `TODO(ghostec): ...`

nit2: Is this really a TODO, or just a plain comment? Do you plan to follow up? TODOs are usually phrase as actions, eg, `// TODO(ghostec): Check if the query can be optimized by grouping ...`.

Reply:
Realistically, I think this is more a note to help improve/fix if this query turns out to be under performing.

Reply:
Please remove the `TODO` and make it a "normal" comment, then.